xbmc/storage/MediaManager.cpp

```diff
@@ -430,7 +430,7 @@ DWORD CMediaManager::GetDriveStatus(const std::string& devicePath)
 }
 
 #ifdef HAS_DVD_DRIVE
-CCdInfo* CMediaManager::GetCdInfo(const std::string& devicePath)
+CCdInfo* CMediaManager::GetCdInfo(const std::string& devicePath, WaitMode waitMode)
```

Comment:
how about implementing `CMediaManager::GetCdInfoLazy(devicePath)` instead of the additional waitMode argument?
though it makes it necessary to swap out following `#ifdeffery TARGET_WINDOWS` to a separate method.
(i just feel a second default argument is odd)

Reply:
Yeah that would be the ideal situation, most likely the generic path already works for Windows too but I don't have a windows Dev environment to validate this at the moment.

Reply:
nah, i mean sth like this (roughly)
```c
CCdInfo* CMediaManager::GetCdInfoWINDOWS_HELPER_OR_WHATEVER(const std::string& devicePath)
{
  if(!m_bhasoptical)
      return NULL; (the full windows stuff from below)
  .
}

CCdInfo* CMediaManager::GetCdInfo(const std::string& devicePath)
{
#ifdef TARGET_WINDOWS
  return GetCdInfoWINDOWS_HELPER_OR_WHATEVER(devicePath);
#else
  return MEDIA_DETECT::CDetectDVDMedia::GetCdInfo(devicePath);
#endif
}

CCdInfo* CMediaManager::GetCdInfoLazy(const std::string& devicePath)
{
#ifdef TARGET_WINDOWS
  return GetCdInfoWINDOWS_HELPER_OR_WHATEVER(devicePath);
#else
  return MEDIA_DETECT::CDetectDVDMedia::GetCdInfoLazy(devicePath);
#endif
}
```
shouldnt change what you have currently. but again only a suggestion. could be it makes it all worse 🤕

Reply:
Yeah it makes it all even worse :)
I've taken a look at the current code and it looks like it's not windows at fault here but it's actually the only platform which uses the proper way of adding the share (using auto sources). The path being changed here is actually sort of deprecated so I think the way forward is to try to improve that first and later on worry with this.
I'm closing this for now, not a big issue anyway and can definitely wait. I'll keep it in mind